src/routes/synapse/handlers.rs

```diff
@@ -11,3 +18,28 @@ pub async fn version(app_data: Data<AppComponents>) -> HttpResponse {
         Err(err_response) => HttpResponse::from_error(err_response),
     }
 }
+
+#[post("/_matrix/client/r0/login")]
+pub async fn login(
+    app_data: Data<AppComponents>,
+    payload: web::Json<SynapseLoginRequest>,
+) -> HttpResponse {
+    match app_data.synapse.login(payload.0).await {
```

Comment:
can we add unit tests for this handler?

Reply:
no complex business logic, there is an integration test which only mocks synapse